node/node.go

```diff
@@ -553,6 +560,44 @@ func createPEXReactorAndAddToSwitch(addrBook pex.AddrBook, config *cfg.Config,
 	return pexReactor
 }
 
+// startStateSync starts an asynchronous state sync process, then switches to fast sync mode.
+func startStateSync(ssR *statesync.Reactor, bcR *bcv0.BlockchainReactor, config *cfg.StateSyncConfig,
+	stateDB dbm.DB, blockStore *store.BlockStore, state sm.State) error {
+	ssR.Logger.Info("Starting state sync")
+
+	lc, err := lite.NewHTTPClient(state.ChainID, lite.TrustOptions{
+		Period: config.TrustedPeriod,
+		Height: config.TrustedHeight,
+		Hash:   config.TrustedHashBytes(),
+	}, config.RPCServers[0], config.RPCServers[1:], litedb.New(dbm.NewMemDB(), ""),
+		// The light client is very chatty, so we use a separate module to silence it at statesync:info.
+		lite.Logger(ssR.Logger.With("module", "lite")))
+	if err != nil {
+		return fmt.Errorf("failed to set up light client: %w", err)
+	}
+
+	go func() {
+		state, commit, err := ssR.Sync(state, lc)
```

Comment:
I think that would couple the reactor too tightly to the other components: we would have to pass the state database, block store, and blockchain reactor into the state sync reactor. I think we should move towards decoupling reactors and components from each other, to reduce system complexity.

Reply:
NOTE: I was talking about `Node#OnStart`. Node object has access to everything

Reply:
Ah, my bad. If I remember correctly I think it was because we don't have the state in `OnStart` - but we can load it, of course. I'll have a second look, I agree it would be better.